File: SNS.Web/src/app/projects/contracts/project-contributor-management.dto.ts
import { InvitingStatus } from '../enums/inviting-status.enum';

export interface ProjectContributorManagementDto {
    contributorRecordId: string;
    profileId: string;
    profileImageUrl: string | null;
    displayName: string;
    specialization: string | null;
    followersCount: number;
    followingCount: number;
    isFollowedByCurrentUser: boolean;
    role: string;
    invitingStatus: InvitingStatus;
    invitationSentAt: string | Date;
    respondedAt: string | Date | null;
    invitationMessage: string | null;
}

File: backend/SNS.Application/Projects/Contracts/ProjectContributorManagementDto.cs
using SNS.Domain.Projects.Enums;

namespace SNS.Application.Projects.Contracts;

public record ProjectContributorManagementDto(
    Guid ContributorRecordId,
    Guid ProfileId,
    string? ProfileImageUrl,
    string DisplayName,
    string? Specialization,
    int FollowersCount,
    int FollowingCount,
    bool IsFollowedByCurrentUser,
    string Role,
    InvitingStatus InvitingStatus,
    DateTime InvitationSentAt,
    DateTime? RespondedAt,
    string? InvitationMessage
);
